### DISCUSSION TITLE
Remove color bar in PPI plot

### DISCUSSION THREAD

**Discussion by ggtt97** [ARM-DOE/pyart#1512] [Q&A] [answered]
Dear all,
Could you tell me how to remove the color bar in the display created using the plot_ppi method?
Thank you in advance.

**Comment by zssherman** [comment, marked as answer]
@ggtt97 Hello! To remove the colorbar, set the parameter in plot_ppi:
colorbar_flag=False

**Reply by ggtt97** [reply]
Thank you!

**Reply by ggtt97** [reply]
Dear @zssherman, I have a small follow-up question. Is it possible to save a single-channel grayscale image? I tried
```
display = pyart.graph.RadarDisplay(radar)
display.plot_ppi("reflectivitity", cmap="gray", title_flag=False, colorbar_flag=False)
plt.axis("off")
plt.savefig("file_name.png")
```
and indeed the saved image looks gray, but it still has 3 channels.

**Reply by zssherman** [reply]
@ggtt97 Good question! If I remember correctly when saving to png or using imsave it might save all 3 channels by default. Not sure though. I did find this that might help though: https://stackoverflow.com/questions/41966514/how-fast-in-python-change-3-channel-rgb-color-image-to-1-channel-gray

Could open the saved image with PIL and convert it, not entirely sure if this is the best route, but you can give it a shot and let me know if it works for you? We can also see if others have any input.

**Reply by ggtt97** [reply]
@zssherman I decided to go with OpenCV to load grayscale images: `cv2.imread(file, cv2.IMREAD_GRAYSCALE)`. That suffices for now. 

By the way do you know a way to directly export PPI data in the form of 2D matrices?

**Reply by zssherman** [reply]
@ggtt97 Sounds good! And I'm not too sure I understand the question. Are you looking to export an array to a text file?

**Reply by ggtt97** [reply]
@zssherman I’d like to get the PPI data that the `plot_ppi` method uses to plot, in the form of numpy arrays.

**Reply by zssherman** [reply]
Oh gotcha, so the data is stored in the fields dictionary in numpy arrays. To access, let's say reflectivity, you would do:

ref_array = radar.fields['reflectivity']['data']

That is the numpy array that stores the data for that field.
